packages/server/api/src/app/app-connection/app-connection-utils.ts

```diff
@@ -1,11 +1,84 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+import { BlockAuthProperty, PropertyType } from '@openops/blocks-framework';
 import {
   AppConnection,
   AppConnectionWithoutSensitiveData,
+  RedactedAppConnection,
 } from '@openops/shared';
 
+const REDACTED_MESSAGE = '**REDACTED**';
+
 export const removeSensitiveData = (
   appConnection: AppConnection,
 ): AppConnectionWithoutSensitiveData => {
   const { value: _, ...appConnectionWithoutSensitiveData } = appConnection;
   return appConnectionWithoutSensitiveData as AppConnectionWithoutSensitiveData;
 };
+
+export function redactSecrets(
+  auth: BlockAuthProperty | undefined,
+  connection: AppConnection,
+): AppConnectionWithoutSensitiveData | RedactedAppConnection {
+  const value: Record<string, any> | undefined = connection.value;
+  if (!auth || !value) {
+    return removeSensitiveData(connection);
+  }
+
+  const redacted = { ...value };
+
+  switch (auth.type) {
+    case PropertyType.SECRET_TEXT: {
```

Comment:
In this case we could technically just use "removeSensitiveData", as the only thing in the properties will be the secret text but idk if its better to stay consistent or not